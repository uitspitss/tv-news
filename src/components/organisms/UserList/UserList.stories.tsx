import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import base from 'paths.macro';

import { UserList, UserListProps } from './UserList';

const path = base.replace(
  /^.+\/(atoms|molecules|organisms|templates|pages)\/(.+)\//,
  '$1/$2',
);

export default {
  title: `${path}`,
  component: UserList,
  args: {
    users: [
      { id: 1, name: 'john' },
      { id: 2, name: 'lisa' },
    ],
  },
} as Meta;

const Template: Story<UserListProps> = (args) => <UserList {...args} />;

export const Default = Template.bind({});
