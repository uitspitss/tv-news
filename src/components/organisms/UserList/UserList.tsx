import React, { memo, VFC } from 'react';
import tw, { styled } from 'twin.macro';

import { User } from '~/models/user';

export type UserListProps = {
  /**
   * class for twin.macro
   */
  className?: string;
  /**
   * user data
   */
  users: User[];
};

type Props = UserListProps;

const Component: VFC<Props> = (props) => {
  const { className, users } = props;

  return (
    <div className={className}>
      {users.map(({ id, name }) => (
        <p key={name}>
          {id}
          {': '}
          {name}
        </p>
      ))}
    </div>
  );
};

const StyledComponent = styled(Component)``;

/**
 * User List
 */
export const UserList = memo((props: UserListProps) => {
  return <StyledComponent {...props} />;
});
