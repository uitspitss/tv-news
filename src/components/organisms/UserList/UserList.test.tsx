import React from 'react';
import { render, screen } from '@testing-library/react';

import { UserList } from './UserList';

describe('organisms/UserList', () => {
  const users = [
    { id: 1, name: 'john' },
    { id: 2, name: 'lisa' },
  ];

  test('snapshot test', () => {
    const { container } = render(<UserList users={users} />);
    expect(container).toMatchSnapshot();
  });

  describe('Behavior', () => {
    beforeEach(() => {
      jest.clearAllMocks();
      render(<UserList users={users} />);
    });

    test('initial view', () => {
      expect(screen.getByText('1: john')).toBeInTheDocument();
      expect(screen.getByText('2: lisa')).toBeInTheDocument();
    });
  });
});
