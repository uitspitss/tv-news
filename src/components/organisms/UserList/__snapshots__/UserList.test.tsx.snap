// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`organisms/UserList snapshot test 1`] = `
<div>
  <div
    class="emotion-0 emotion-1"
  >
    <p>
      1
      : 
      john
    </p>
    <p>
      2
      : 
      lisa
    </p>
  </div>
</div>
`;
